### PR TITLE
Limit telemetry payload size to 65K

### PR DESCRIPTION
**Pull request by jaer-tsun**
**What this PR does / why we need it**:
The previous payload limit of 2MB exceeded the server limit, therefore, this change will remedy that by reducing the max payload size to 65535 bytes.

### CODE REVIEW

File: telemetry/telemetrybuffer.go [hunk 1]
```diff
@@ -253,25 +253,27 @@ func (pl *Payload) push(x interface{}) {
...
-		case NPMReport:
-			npmReport := x.(NPMReport)
-			npmReport.Metadata = metadata
-			pl.NPMReports = append(pl.NPMReports, npmReport)
-		case CNSReport:
-			cnsReport := x.(CNSReport)
-			cnsReport.Metadata = metadata
-			pl.CNSReports = append(pl.CNSReports, cnsReport)
-		}
+	switch x.(type) {
```
Comment (tamilmani1989) [inline telemetry/telemetrybuffer.go]: why did we change this and introduced capPayload function?

File: telemetry/telemetry_test.go [hunk 2]
```diff
@@ -155,3 +178,18 @@ func TestSetReportState(t *testing.T) {
 		t.Errorf("Error removing telemetry file due to %v", err)
 	}
 }
+
+func TestPayloadCap(t *testing.T) {
```
Comment (tamilmani1989) [inline telemetry/telemetry_test.go]: can we also have a test case for exceeding max cap?

File: telemetry/telemetrybuffer.go [hunk 3]
```diff
@@ -285,11 +286,34 @@ func (pl *Payload) reset() {
...
+	case NPMReport:
+		reportType = npm
+		json.NewEncoder(&body).Encode(x.(NPMReport))
+	case CNSReport:
+		reportType = cns
+		json.NewEncoder(&body).Encode(x.(CNSReport))
+	}
+
+	updatedPayloadSize := uint16(body.Len()) + payloadSize
+	if notExceeded = updatedPayloadSize < MaxPayloadSize && payloadSize < updatedPayloadSize; notExceeded {
```
Comment (tamilmani1989) [inline telemetry/telemetrybuffer.go]: payloadSize < updatedPayloadSize -> why do we need this check?

File: telemetry/telemetrybuffer.go [hunk 4]
```diff
@@ -22,20 +22,24 @@ import (
...
+	logName                   = "azure-vnet-telemetry"
+	MaxPayloadSize     uint16 = 65535
+	dnc                       = "DNC"
+	cns                       = "CNS"
+	npm                       = "NPM"
+	cni                       = "CNI"
 )
 
 var telemetryLogger = log.NewLogger(logName, log.LevelInfo, log.TargetStderr)
+var payloadSize uint16 = 0
```
Comment (tamilmani1989) [inline telemetry/telemetrybuffer.go]: can't this be part of payload structure instead of global?

File: telemetry/telemetrybuffer.go (hunk 3 above)
Comment (jaer-tsun) [reply telemetry/telemetrybuffer.go]: in case of overflow

File: telemetry/telemetrybuffer.go (hunk 4 above)
Comment (jaer-tsun) [reply telemetry/telemetrybuffer.go]: let's keep payload the same as contract on host

**Pull request merged**